src/modules/Chat.ts

```diff
@@ -0,0 +1,44 @@
+import { User } from "src/modules/User";
+import { Socket } from "socket.io-client";
+
+interface chatMessage {
+  timestamp: number;
+  sender: string;
+  message: string;
+}
+export class Chat {
+  private user: User;
+  private roomId: string;
+  private channel: Socket;
+
+  constructor(user: User, roomId: string, channel: Socket) {
+    this.user = user;
+    this.roomId = roomId;
+    this.channel = channel;
+  }
+
+  sendMessage(message: string) {
+    this.channel.emit("postChatMessage", {
+      message,
+      roomId: this.roomId,
+      user: this.user?.getToken(),
+    });
+  }
+
+  fetchMessages(): Promise<chatMessage[]> {
+    this.channel.emit("requestAllChatMessages", {
+      roomId: this.roomId,
+    });
+    return new Promise<chatMessage[]>((resolve, reject) => {
+      this.channel.on(
+        "fetchAllMessages",
+        (messages: chatMessage[], errorMessage: string) => {
```

Comment:
By the time the socket responds back the user could be inside another room. We need to send back the `roomId` with the message list.

Reply:
But the user will already have `roomId` when he creates `Chat` instance. Why does `fetchMessages` need to send back `roomId` ?

Reply:
Agreed, I read about `room`s in socket.io documentation and we should be good w/o 'roomId` here.